### PR TITLE
[PM-10877] Use new version calculation in CI-main workflow 

### PR DESCRIPTION
**Pull request by vvolkgang**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10877

## 📔 Objective

Implement the new version calculation in `CI-main.yml`. Testing this brought the need to trigger builds without distributing them, added a new input to help managing that. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/CI-main.yml [hunk 1]
```diff
@@ -24,72 +27,61 @@ jobs:
...
-            echo "  Latest Minor Version: $latest_minor_version"
-            echo "  Latest Patch Version: $latest_patch_version"
+      - name: Read version info
+        id: version_info
+        run: |
+          # test if dispatch-version was skipped. In that case, creates the same .json file expected by the Upload artifact step
+          if [ ! -f version-info/version_info.json ]; then
+            json='{
+              "version_number": "${{ inputs.build-number }}",
+              "version_name": "${{ inputs.build-version }}"
```
Comment (fedemkr) [inline .github/workflows/CI-main.yml:59]: 🤔 If `inputs.build-number` doesn't have value but `inputs.build-version` has value then `Calculate version` is not triggered, therefore which would be the value here for `version_number`? Could there be any repercussions on this scenario? What about the inverse?

Comment (vvolkgang) [reply .github/workflows/CI-main.yml:62]: @fedemkr Changed how this works, that scenario should be covered.

File: .github/workflows/CI-main.yml [hunk 2]
```diff
@@ -12,84 +12,84 @@ on:
...
       matrix:
         variant: [Beta, Production]
     with:
       build-variant: ${{ matrix.variant }}
-      build-version: ${{ needs.resolve-values.outputs.build_version }}
-      build-number: ${{ needs.resolve-values.outputs.build_number }}
+      build-version: ${{ needs.resolve-values.outputs.version_name }}
+      build-number: ${{ needs.resolve-values.outputs.version_number }}
       xcode-version: ${{ needs.resolve-values.outputs.xcode_version }}
+      distribute: ${{ fromJSON(needs.resolve-values.outputs.distribute_to_testflight) }}
```
Comment (vvolkgang) [inline .github/workflows/CI-main.yml:93]: 💡 outputs are always strings, `fromJSON` is converting it to a boolean

File: .github/workflows/CI-main.yml [hunk 3]
```diff
@@ -12,84 +12,84 @@ on:
...
     strategy:
       matrix:
         variant: [Beta, Production]
     with:
       build-variant: ${{ matrix.variant }}
-      build-version: ${{ needs.resolve-values.outputs.build_version }}
-      build-number: ${{ needs.resolve-values.outputs.build_number }}
+      build-version: ${{ needs.resolve-values.outputs.version_name }}
+      build-number: ${{ needs.resolve-values.outputs.version_number }}
       xcode-version: ${{ needs.resolve-values.outputs.xcode_version }}
```
Comment (vvolkgang) [inline .github/workflows/CI-main.yml:92]: for some reason `${{ env.*** }}` isn't available here, we can access a bunch of other contexts, just not `env`. Seems like an oversight. We'll need to keep routing these through another job for the time being.

**Review by fedemkr — APPROVED** [review]

**Pull request merged**